package.json

```diff
@@ -30,6 +30,7 @@
     },
     "scripts": {
         "lint": "eslint . --report-unused-disable-directives",
+        "postinstall": "node tests/patch-esmock.js",
```

Comment:
Does `postinstall` script also run when user installs this package?

Reply:
good catch! 👍 

I think we need to use `prepare` - it won't run in that case.

Reply:
> Does `postinstall` script also run when user installs this package?

I think you mean when a user runs `npm install @eslint/create-config` in a different package folder, or when they run `npm install` in a different package that includes `@eslint/create-config` as a dependency or devDependency. In either case: I don't know 🤷 and I don't trust the [npm docs](https://docs.npmjs.com/cli/v10/using-npm/scripts) too much in this regard.

I have a package with some scripts I use for testing exactly for those cases: https://unpkg.com/@fasttime/npm-test@0.0.0/package.json. Now if I run `npm install @fasttime/npm-test` in the directory of another package, none of the scripts seems to be run. I would assume this is the behavior we should expect from this change as well, but it's clearly possible that I'm missing something.

Reply:
I believe the scripts are run in sort of silent mode. When I try with `--loglevel verbose`, it looks like some of the scripts (including `postinstall`) are run?

```
$ npm install --loglevel verbose @fasttime/npm-test
npm verb cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
npm info using npm@10.1.0
npm info using node@v20.7.0
npm verb title npm install @fasttime/npm-test
npm verb argv "install" "--loglevel" "verbose" "@fasttime/npm-test"
npm verb logfile logs-max:10 dir:C:\Users\milos\AppData\Local\npm-cache\_logs\2023-10-24T12_05_00_631Z-
npm verb logfile C:\Users\milos\AppData\Local\npm-cache\_logs\2023-10-24T12_05_00_631Z-debug-0.log
npm http fetch GET 200 https://registry.npmjs.org/@fasttime%2fnpm-test 1413ms (cache revalidated)
npm http fetch GET 200 https://registry.npmjs.org/inherits 5ms (cache hit)
npm info run @fasttime/npm-test@0.0.0 preinstall node_modules/@fasttime/npm-test echo *** preinstall ***
npm info run @fasttime/npm-test@0.0.0 preinstall { code: 0, signal: null }
npm info run @fasttime/npm-test@0.0.0 install node_modules/@fasttime/npm-test echo *** install ***
npm info run @fasttime/npm-test@0.0.0 install { code: 0, signal: null }
npm info run @fasttime/npm-test@0.0.0 postinstall node_modules/@fasttime/npm-test echo *** postinstall ***
npm info run @fasttime/npm-test@0.0.0 postinstall { code: 0, signal: null }
npm http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 222ms

added 2 packages, and audited 3 packages in 2s

found 0 vulnerabilities
npm verb exit 0
npm info ok
```

Reply:
Interesting, I never noticed that npm was silencing the output. In this case, yes, we need to use the `prepare` script instead of `postinstall`. Now, `prepare` also runs on `npm pack` and `npm publish` when no dependencies are being installed, so we will also need a check to ensure that nothing happens if `esmock_legacy` is missing. I'll _prepare_ an update.

Reply:
Updated, please check again.